python/ray/tune/scripts.py

```diff
@@ -0,0 +1,185 @@
+    '--sort', default=None, type=str, help='Select which column to sort on.')
+def list_trials(experiment_path, sort):
+    _list_trials(experiment_path, sort)
+
+
+def _list_experiments(project_path, sort, info_keys=DEFAULT_PROJECT_INFO_KEYS):
+    base, experiment_paths, _ = list(os.walk(project_path))[0]  # clean this
+
+    experiment_data_collection = []
+    for experiment_path in experiment_paths:
+        experiment_state_path = glob.glob(
+            os.path.join(base, experiment_path, "experiment_state*.json"))
+
+        if not experiment_state_path:
+            # TODO(hartikainen): Print some warning?
+            continue
+
+        with open(experiment_state_path[0]) as f:
+            experiment_state = json.load(f)
+
+        checkpoints = pd.DataFrame(experiment_state["checkpoints"])
+        runner_data = experiment_state["runner_data"]
+        timestamp = experiment_state.get("timestamp")
+
+        experiment_data = {
+            "name": experiment_path,
+            "start_time": runner_data.get("_start_time"),
+            "timestamp": datetime.fromtimestamp(timestamp)
+            if timestamp else None,
+            "total_trials": checkpoints.shape[0],
+            "running_trials": (checkpoints["status"] == Trial.RUNNING).sum(),
+            "terminated_trials": (
+                checkpoints["status"] == Trial.TERMINATED).sum(),
+            "error_trials": (checkpoints["status"] == Trial.ERROR).sum(),
+        }
+        experiment_data_collection.append(experiment_data)
+
+    info_df = pd.DataFrame(experiment_data_collection)[list(info_keys)]
+    if sort:
+        info_df = info_df.sort_values(by=sort)
+        info_df = info_df.reset_index(drop=True)
+
+    print_df = pd.DataFrame()
+    columns = list(info_keys)
+    dropped = []
+    empty = []
+    # column display priority is based on the info_keys passed in
+    for i in range(len(columns)):
+        print_df[columns[i]] = info_df[columns[i]]
+        table = tabulate(print_df, headers="keys", tablefmt="psql")
+        if info_df[columns[i]].isnull().all():
+            empty.append(columns[i])
+            print_df.drop(columns[i], axis=1, inplace=True)
+        elif str(table).index('\n') > TERMINAL_WIDTH:
+            print_df.drop(columns[i], axis=1, inplace=True)
+            table = tabulate(print_df, headers="keys", tablefmt="psql")
+            dropped += columns[i:]
+            break
+
+    print(table)
+    if dropped:
+        print("Dropped columns:", dropped)
+        print("Please increase your terminal size to view remaining columns.")
+    if empty:
+        print("Empty columns:", empty)
+
+
+@cli.command()
+@click.argument("project_path", required=True, type=str)
+@click.option(
+    '--sort', default=None, type=str, help='Select which column to sort on.')
+def list_experiments(project_path, sort):
+    _list_experiments(project_path, sort)
+
+
+cli.add_command(list_trials, name="ls")
+cli.add_command(list_experiments, name="lsx")
```

Comment:
I think it would make sense to also have non-aliased names for these:
```python
cli.add_command(list_trials, name="list-trials")
cli.add_command(list_trials, name="ls")
cli.add_command(list_experiments, name="list-experiments")
cli.add_command(list_experiments, name="lsx")
```

Reply:
Added for now but I suspect this is optional (still need to build ray and test this)

Reply:
```
╰─ python scripts.py
Usage: scripts.py [OPTIONS] COMMAND [ARGS]...

Options:
  --help  Show this message and exit.

Commands:
  list-experiments  Lists experiments in the directory subtree.
  list-trials       Lists trials in the directory subtree...
  list_experiments  Lists experiments in the directory subtree.
  list_trials       Lists trials in the directory subtree...
  ls                Lists trials in the directory subtree...
  lsx               Lists experiments in the directory subtree.
```

By default, it already registers `list_experiments` and `list_trials` so I think we can forgo the extra alias?

Reply:
Ah, agreed. I forgot about the default ones.